mlir/lib/Driver/CompilerDriver.cpp

```diff
@@ -360,8 +374,53 @@ LogicalResult inferMLIRReturnTypes(MLIRContext *ctx, llvm::Type *returnType,
     return failure();
 }
 
-LogicalResult runLLVMPasses(const CompilerOptions &options,
-                            std::shared_ptr<llvm::Module> llvmModule, CompilerOutput &output)
+LogicalResult runCoroLLVMPasses(const CompilerOptions &options,
+                                std::shared_ptr<llvm::Module> llvmModule, CompilerOutput &output)
+{
+
+    auto &outputs = output.pipelineOutputs;
+
+    // Create a pass to lower LLVM coroutines (similar to what happens in O0)
+    llvm::ModulePassManager CoroPM;
+    CoroPM.addPass(llvm::CoroEarlyPass());
+    llvm::CGSCCPassManager CGPM;
+    CGPM.addPass(llvm::CoroSplitPass());
+    CoroPM.addPass(llvm::createModuleToPostOrderCGSCCPassAdaptor(std::move(CGPM)));
+    CoroPM.addPass(llvm::CoroCleanupPass());
+    CoroPM.addPass(llvm::GlobalDCEPass());
+
+    // Create the analysis managers.
+    llvm::LoopAnalysisManager LAM;
+    llvm::FunctionAnalysisManager FAM;
+    llvm::CGSCCAnalysisManager CGAM;
+    llvm::ModuleAnalysisManager MAM;
+
+    llvm::PassBuilder PB;
+    PB.registerModuleAnalyses(MAM);
+    PB.registerCGSCCAnalyses(CGAM);
+    PB.registerFunctionAnalyses(FAM);
+    PB.registerLoopAnalyses(LAM);
+    PB.crossRegisterProxies(LAM, FAM, CGAM, MAM);
+
+    // Create the pass manager.
+    llvm::ModulePassManager MPM;
+    MPM.addPass(llvm::CoroConditionalWrapper(std::move(CoroPM)));
```

Comment:
I was thinking the second pass manager seems kind of redundant, by I guess it exists to condition the execution of the coro passes using the wrapper. We might be able to skip the wrapper and second pass manager, given that we know there are coroutines present when the option is enabled (unless the user enabled it with no qnodes present).

Reply:
Yes correct, I forgot to remove it after adding the async to the option

Reply:
Done